src/sparsezoo/v2/model_directory.py

```diff
@@ -0,0 +1,502 @@
+            elif isinstance(file, dict):
+                raise NotImplementedError()
+
+        return all(validations.values())
+
+    def analyze(self):
+        # TODO: This will be the onboarding task for Kyle
+        pass
+
+    def generate_outputs(
+        self, engine_type: str
+    ) -> Union[List[np.ndarray], typing.OrderedDict[str, np.ndarray]]:
+        """
+        Chooses the appropriate engine type to load the onnx model.
+        Then, feeds the data (sample original inputs)
+        to generate model outputs (in the iterative fashion).
+        :params engine_type: name of the inference engine
+        :returns engine output.
+        """
+        if engine_type not in ENGINES:
+            raise ValueError(f"The argument `engine_type` must be one of {ENGINES}")
+        elif engine_type == "onnxruntime":
+            for output in self._run_with_onnx_runtime():
+                yield output
+        else:
+            for output in self._run_with_deepsparse():
+                yield output
+
+    def _get_directory(
+        self,
+        file: Dict[str, Any],
+        directory_class: Directory,
+        display_name: Optional[str] = None,
+        regex: Optional[bool] = False,
+    ) -> Union[Directory, None]:
+        # Takes a file dictionary and returns a Directory() object, if successful.
+        # Optionally, can do:
+        #   string matching (if `display_name` not None)
+        #   regex (if `display_name` not None and `regex`)
+        #   file validation (if file's path is not None)
+
+        match = True
+        if display_name:
+            if regex:
+                pattern = re.compile(display_name)
+                match = re.search(pattern, file["display_name"])
+
+            else:
+                match = display_name == file["display_name"]
+
+        if not match:
+            logging.warning(
+                "Could not find a directory with "
+                f"display_name / regex_pattern: {display_name}"
+            )
+            return None
+
+        name, path, url = file.get("display_name"), file.get("path"), file.get("url")
+
+        # directory is a tar file
+        if self._is_file_tar(file):
+            directory = directory_class(files=[], name=name, path=path, url=url)
+            return directory
+
+        # directory is folder
+        else:
+            # is directory using the 'contents' key.
+            # this is a placeholder for the nested files that
+            # the directory contains
+            if file.get("contents"):
+                files_within = file["contents"]
+
+            # is directory locally on the machine
+            else:
+                paths_within = glob.glob(os.path.join(path, "*"))
+                files_within = [
+                    file_dictionary(display_name=os.path.basename(path), path=path)
+                    for path in paths_within
+                ]
```

Comment:
nit:
```suggestion
                files_within = (
                    file_dictionary(display_name=os.path.basename(path), path=path)
                    for path in paths_within
                )
```
Can be converted to a generator again to `yield` values only when needed